test/e2e/plain_provisioner_test.go

```diff
@@ -494,6 +497,130 @@ var _ = Describe("plain provisioner bundle", func() {
+				var (
+					sa  corev1.ServiceAccount
+					crb rbacv1.ClusterRoleBinding
+					job batchv1.Job
+					pod corev1.Pod
+				)
+				By("eventually reporting an Unpacked phase", func() {
+					Eventually(func() (string, error) {
+						if err := c.Get(ctx, client.ObjectKeyFromObject(bundle), bundle); err != nil {
+							return "", err
+						}
+						return bundle.Status.Phase, nil
+					}).Should(Equal(rukpakv1alpha1.PhaseUnpacked))
+				})
+
+				By("eventually writing a content URL to the status", func() {
+					Eventually(func() (string, error) {
+						err := c.Get(ctx, client.ObjectKeyFromObject(bundle), bundle)
+						Expect(err).To(BeNil())
+						return bundle.Status.ContentURL, nil
+					}).Should(Not(BeEmpty()))
+				})
+
+				By("creating a pod with right authority", func() {
+					// Create a temporary ServiceAccount
+					sa = corev1.ServiceAccount{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "rukpak-svr-sa",
+							Namespace: defaultSystemNamespace,
+						},
+					}
+					err := c.Create(ctx, &sa)
+					Expect(err).To(BeNil())
+
+					// Create a temporary ClusterRoleBinding to bind the ServiceAccount to bundle-reader ClusterRole
+					crb = rbacv1.ClusterRoleBinding{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "rukpak-svr-crb",
+							Namespace: defaultSystemNamespace,
+						},
+
+						Subjects: []rbacv1.Subject{{Kind: "ServiceAccount", Name: "rukpak-svr-sa", Namespace: defaultSystemNamespace}},
+						RoleRef:  rbacv1.RoleRef{APIGroup: "rbac.authorization.k8s.io", Kind: "ClusterRole", Name: "bundle-reader"},
+					}
+
+					err = c.Create(ctx, &crb)
+					Expect(err).To(BeNil())
+					url := bundle.Status.ContentURL
+
+					// Create a Job that reada from the URL and outputs contents in the pod log
+					mounttoken := true
+					job = batchv1.Job{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "rukpak-svr-job",
+							Namespace: defaultSystemNamespace,
+						},
+						Spec: batchv1.JobSpec{
+							Template: corev1.PodTemplateSpec{
+								Spec: corev1.PodSpec{
+									Containers: []corev1.Container{
+										{
+											Name:    "rukpak-svr",
+											Image:   "curlimages/curl",
+											Command: []string{"sh", "-c", "curl -sSLk -H \"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\" -o - " + url + " | tar ztv"},
+										},
+									},
+									ServiceAccountName:           "rukpak-svr-sa",
+									RestartPolicy:                "Never",
+									AutomountServiceAccountToken: &mounttoken,
+								},
+							},
+						},
+					}
+					err = c.Create(ctx, &job)
+					Expect(err).To(BeNil())
+					Eventually(func() (bool, error) {
+						err = c.Get(ctx, types.NamespacedName{Name: "rukpak-svr-job", Namespace: defaultSystemNamespace}, &job)
+						if err != nil {
+							return false, err
+						}
+						return job.Status.CompletionTime != nil && job.Status.Succeeded == 1, err
+					}).Should(BeTrue())
+				})
```

Comment:
Move this to a BeforeEach clause?

Reply:
Thanks for the idea.  It can clean up the resources at test failure now.